# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/madhusowmyanaidu/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Lane:** Search visibility and content performance
I want to investigate whether page-level search and content characteristics are associated with better search performance. I am particularly interested in understanding whether factors such as average position, impressions, click-through rate, and content age show useful relationships with page performance. This question is useful because it could help prioritize which pages deserve attention first. I will treat the findings as observed and directional rather than claiming causation.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**Decision:** Which pages should be prioritized for SEO/content review?
**Action:** An SEO or content team could use the analysis to decide which pages deserve investigation or optimization first.
**Cost of a wrong call:** Prioritizing the wrong pages could waste limited time and resources, while failing to prioritize a genuinely promising page could delay potential improvements. Therefore, the analysis should be treated as decision support rather than a guaranteed prediction of search-engine behavior.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [5]:
!git clone https://github.com/madhusowmyanaidu/flyrank-ml-internship.git
import os
import pandas as pd

os.chdir("/content/flyrank-ml-internship")

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Number of pages:", len(df))
print("Median impressions_90d:", df["impressions_90d"].median())
print("Median avg_position:", df["avg_position"].median())
print("Median ctr:", df["ctr"].median())


Cloning into 'flyrank-ml-internship'...
remote: Enumerating objects: 126, done.
remote: Counting objects: 100% (126/126), done.
remote: Compressing objects: 100% (82/82), done.
remote: Total 126 (delta 38), reused 97 (delta 28), pack-reused 0 (from 0)
Receiving objects: 100% (126/126), 1.87 MiB | 8.65 MiB/s, done.
Resolving deltas: 100% (38/38), done.
Number of pages: 30000
Median impressions_90d: 731.0
Median avg_position: 10.8
Median ctr: 0.07


**Data snapshot:** The starter dataset contains 30,000 pages. The median number of impressions over the last 90 days is 731, while the median average search position is 10.8 and the median CTR is 0.07. These numbers show that the dataset contains substantial variation in page-level search visibility and performance, making it useful for investigating which page characteristics are associated with search performance.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What I can claim: I can identify observed relationships and directional patterns between page-level characteristics and search performance. These findings can be used as decision support to help prioritize pages for further SEO or content investigation.

**What I cannot claim:** I cannot prove that one factor causes another, and I cannot claim to predict or explain Google's ranking algorithm. Other factors may influence the observed relationships. Therefore, my conclusions will be presented as observed and directional findings rather than causal proof.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.